ddtrace/tracer/remote_config.go

```diff
@@ -298,13 +299,26 @@ func initalizeDynamicInstrumentationRemoteConfigState() {
 			time.Sleep(time.Second * 5)
 			diRCState.Lock()
 			for _, v := range diRCState.state {
+				accessStringsToMitigatePageFault(v.runtimeID, v.configPath, v.configContent)
 				passProbeConfiguration(v.runtimeID, v.configPath, v.configContent)
 			}
 			diRCState.Unlock()
 		}
 	}()
 }
 
+func accessStringsToMitigatePageFault(strs ...string) {
```

Comment:
I'm less worried about the function itself being optimized away (it doesn't seem to do it according to the disassembler), and more about the actual lines that access the variables. I observed the behavior this PR is trying to mitigate over a large amount of time across different services (though admittedly only on 1 environment), and in fact, we still have occasional misses (seem to typically be followed by hits). 

As a result I simplified the access to writing the strings to /dev/null and in fact this removed any amount of misses in my manual testing. It also simplifies the code. So i've pushed this change.

Reply:
Ok i've further updated this to just use io.Discard

Reply:
Could [`runtime.KeepAlive`](https://pkg.go.dev/runtime#KeepAlive) be enough here? :thinking:

Reply:
@eliottness I wasn't aware of this function. Looking at the code it just performs a simple `println()` on the passed value which is what we're doing except we won't print the config to our customers service logs.

The theory we're accepting is that the string addresses need to be in the core-specific TLB before the hooked function is called.

Reply:
> I wasn't aware of this function. Looking at the code it just performs a simple println() on the passed value which is what we're doing except we won't print the config to our customers service logs.

@grantseltzer  I probably badly explained what I was suggesting here: Yes technically there is a call to `println` in there but it's never reach because the `if` statement end up being always false. Except the compiler cannot know if the variable will stay false so it cannot optimize the if statement away. In the end, it ends up running just a few instructions where as `io.Discard([]byte(str))` does a copy of the string and allocates it on the heap to send it to the io package. In the PR description you where talking about the "critical path for Go DI" so I thought you would welcome a performance boost but now that I think of it it's probably unnecessary. Feel free to take or discard (pun intended) my comment

Reply:
@eliottness Ahh, I see. Let me test to make sure this works at resolving this issue, and would rather use this instead. But yea performance isn't the focus of concern considering it gets called once every 5 seconds.